src/sigstore.ts

```diff
@@ -66,30 +78,42 @@ export async function signAttestation(
   options: SignOptions = {}
 ): Promise<Bundle> {
   const fulcio = new Fulcio({ baseURL: options.fulcioBaseURL });
-  const rekor = new Rekor({ baseURL: options.rekorBaseURL });
+  const tlog = newTLogClient(options);
   const idps = configureIdentityProviders(options);
   const signer = new Signer({
     fulcio,
-    rekor,
+    tlog,
     identityProviders: idps,
   });
 
   const bundle = await signer.signAttestation(payload, payloadType);
-  return BundleSerializer.toJSON(bundle) as Bundle;
+  return bundleToJSON(bundle) as Bundle;
 }
 
 export async function verify(
   bundle: Bundle,
   data?: Buffer,
   options: VerifierOptions = {}
 ): Promise<boolean> {
-  const rekor = new Rekor({ baseURL: options.rekorBaseURL });
-  const verifier = new Verifier({ rekor });
+  const tlog = newTLogClient(options);
+  const verifier = new Verifier({ tlog });
 
-  const b = BundleSerializer.fromJSON(bundle);
+  const b = bundleFromJSON(bundle);
   return verifier.verify(b, data);
 }
 
+// Accepts a signed DSSE envelope and a PEM-encoded public key to be added to the
+// transparency log. Returns a Sigstore bundle suitable for offline verification.
+export async function addEnvelopeToTransparencyLog(
+  envelope: Envelope,
+  key: string,
+  options: SignOptions = {}
+): Promise<Bundle> {
+  const tlog = newTLogClient(options);
+  const bundle = await tlog.createDSSEEntry(envelope, key);
+  return bundleToJSON(bundle) as Bundle;
+}
+
```

Comment:
Encapsulating all the Rekor logic within the `TLog` class means we can easily reuse the implementation both here and in the Signer.

Reply:
Open to suggestions for naming this function. Don't love what I've come up with here but struggled to find something better.

Reply:
Yeah not sure I have many better ideas but any reason for not calling out rekor here, e.g:
- `createRekorEntryFromDsseEnvelope`
- addDsseEnvelopeToRekor`

Reply:
I sorta had it in my head that "Rekor" was an implementation detail that we didn't want to expose . . . but I'm probably overthinking it. This is the SIGSTORE library after all.

I think I'll go with `createRekorEntry`